src/main/java/org/jabref/gui/entryeditor/citationrelationtab/CitationRelationsTab.java

```diff
@@ -270,6 +295,40 @@ private void styleFetchedListView(CheckListView<CitationRelationItem> listView)
         listView.setSelectionModel(new NoSelectionModel<>());
     }
 
+    private String getSourceString(BibEntry entry, BibDatabaseMode type) throws IOException {
+        StringWriter writer = new StringWriter();
+        BibWriter bibWriter = new BibWriter(writer, OS.NEWLINE);
+        FieldWriter fieldWriter = FieldWriter.buildIgnoreHashes(this.preferences.getFieldPreferences());
+        new BibEntryWriter(fieldWriter, new BibEntryTypesManager()).write(entry, bibWriter, type);
```

Comment:
Can you try to get the `BibEntryTypesManager` be passed? It should be available "somewhere" in the call hierarchy. `org.jabref.gui.collab.entrychange.PreviewWithSourceTab` also has it.

Reply:
Please add an `@implNote` that this code is similar to `PreviewWithSourceTab#getSourceString`. 

@Siedlerchr I think it is OK to have code duplication here. Alternatively, we could introduce `org.jabref.logic.exporter.BibEntryPrinter` having a single method `getAsString`.

Reply:
Duplicated code is imho okay, just the EntryTypesManaager as constructor parameter